### PR TITLE
refactor(tasks): extract + test helper from assignee list

### PR DESCRIPTION
**Pull request by alexkrolick**
Followup for #1587

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @alexkrolick has signed the CLA. Thanks for the pull request!

File: src/elements/content-sidebar/activity-feed/task-new/AssigneeDetails.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
+// @flow strict
+import * as React from 'react';
+import { FormattedMessage } from 'react-intl';
+import ReadableTime from '../../../../components/time/ReadableTime';
+import commonMessages from '../../../common/messages';
+import messages from './messages';
+import { TASK_NEW_APPROVED, TASK_NEW_REJECTED, TASK_NEW_COMPLETED, TASK_NEW_NOT_STARTED } from '../../../../constants';
+import type { ISODate } from '../../../../common/types/core';
+
+import './AssigneeList.scss';
```
Comment (jstoffan) [inline src/elements/content-sidebar/activity-feed/task-new/AssigneeDetails.js]: Should this have its own stylesheet, instead?

**Comment by alexkrolick** [comment]
For some reason the local prepush lint is not catching things our CI is complaining about...

File: src/elements/content-sidebar/activity-feed/task-new/__tests__/AssigneeDetails-test.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    test.each`
+        status                  | completedAt
+        ${TASK_NEW_APPROVED}    | ${MOCK_DATE_STRING}
+        ${TASK_NEW_REJECTED}    | ${MOCK_DATE_STRING}
+        ${TASK_NEW_NOT_STARTED} | ${MOCK_DATE_STRING}
+        ${'invalid_string'}     | ${MOCK_DATE_STRING}
+        ${TASK_NEW_APPROVED}    | ${MOCK_DATE}
+        ${null}                 | ${MOCK_DATE}
+    `('should render details for status $status', ({ status, completedAt }) => {
+        const wrapper = mount(<AssigneeDetails status={status} user={MOCK_USER} completedAt={completedAt} />);
```
Comment (tjgupta) [inline src/elements/content-sidebar/activity-feed/task-new/__tests__/AssigneeDetails-test.js]: Can we use `shallow` here? Or does it have to use `mount` for some reason?

File: src/elements/content-sidebar/activity-feed/task-new/__tests__/__snapshots__/AssigneeDetails-test.js.snap [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+      </FormattedMessage>
+    </div>
+  </div>
+</Component>
+`;
+
+exports[`elements/content-sidebar/ActivityFeed/task-new/AssigneeDetails should render details for status invalid_string 1`] = `
+<Component
+  completedAt="2019-01-01T00:00:000Z"
+  status="invalid_string"
```
Comment (tjgupta) [inline src/elements/content-sidebar/activity-feed/task-new/__tests__/__snapshots__/AssigneeDetails-test.js.snap]: Is this a valuable snapshot? Should it not render the component if the status is invalid or something other behavior?

File: src/elements/content-sidebar/activity-feed/task-new/__tests__/__snapshots__/AssigneeDetails-test.js.snap [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+      user one
+    </div>
+  </div>
+</Component>
+`;
+
+exports[`elements/content-sidebar/ActivityFeed/task-new/AssigneeDetails should render details for status null 1`] = `
+<Component
+  completedAt={2019-01-01T00:00:00.000Z}
+  status={null}
```
Comment (tjgupta) [inline src/elements/content-sidebar/activity-feed/task-new/__tests__/__snapshots__/AssigneeDetails-test.js.snap]: Same here.... valuable?

**Pull request merged**